### PR TITLE
obs_info should handle identity obs without crashing

### PR DESCRIPTION
**Pull request by nancycollins**
made a separate array to store the info for identity obs
rather than try to support negative indices in the oinfo array.

also fix time string formatting if running with no calendar.

### CODE REVIEW

File: assimilation_code/programs/obs_utils/obs_info.f90
Comment (timhoar) [inline assimilation_code/programs/obs_utils/obs_info.f90:231]: why did you specify a fixed width?  get_name_for_type_of_obs() returns a length 31 string, so this seems unnecessary.

File: assimilation_code/programs/obs_utils/obs_info.f90
Comment (timhoar) [inline assimilation_code/programs/obs_utils/obs_info.f90:248]: This is the block that reports on the summary of the identity observations.

File: assimilation_code/programs/obs_utils/obs_info.f90
Comment (timhoar) [inline assimilation_code/programs/obs_utils/obs_info.f90:241]: To get them to line up nicely, the string has to have spaces added to make the string 32 chars wide.
```suggestion
                                             "IDENTITY_OBSERVATIONS           "//", ", &
```

**Review by timhoar — APPROVED** [review]
When I combined all the example obs_seq.out files in the lorenz_63 directory and ran it through obs_info, everything worked but the formatting was a little sub-optimal. Not enough to request a change IMO, but this is what it looks like:
```
 Totals for all obs types:
   Count:         2600
 ---------------------------------------------------------
 RAW_STATE_VARIABLE                  600
           IDENTITY_OBSERVATIONS    2000
```

note that the strings are not left-justified.

**Review by hkershaw-brown — APPROVED** [review]
Looks good.
I agree with Tim's suggestion for changing the printed output. 

Not required for this pull request, 
Is there any documentation for the code in DART/observations/utilities/ ? For example, when would you use the obs_sequence_tool vs one of the smaller obs_blah_blah_blah programs. 
Should there be observation utilities for each of the location mods.  There is only oned and threed_sphere at the moment:

   channel           
   threed            
   **threed_sphere**     
  twod_annulus     
   annulus           
  column           
  **oned**              
  threed_cartesian 
  twod              
  twod_sphere

**Comment by nancycollins** [comment]
i also agree with tim's comments and plan to make another update to this code, so please don't merge it yet.  thanks!

**Comment by nancycollins** [comment]
i know you're waiting to edit out-of-date content until after this is on the master branch, but this first page: 

 https://dart-documentation.readthedocs.io/en/latest/README.html

says Manhattan is new and only supports a subset of the models.  it's not new, and really it supports most models people are actually running so i'd argue the subset part is unnecessarily frightening.  it also tells users to fork the repo first thing instead of just doing a clone.  are any of those things serious enough we want to fix them now, or just put them on the list for later?  (if now, i can take a shot at doing minor updates to it.  just let me know.)

**Comment by hkershaw-brown** [comment]
> i know you're waiting to edit out-of-date content until after this is on the master branch, but this first page:
> 
> https://dart-documentation.readthedocs.io/en/latest/README.html
> 
> says Manhattan is new and only supports a subset of the models. it's not new, and really it supports most models people are actually running so i'd argue the subset part is unnecessarily frightening. it also tells users to fork the repo first thing instead of just doing a clone. are any of those things serious enough we want to fix them now, or just put them on the list for later? (if now, i can take a shot at doing minor updates to it. just let me know.)

I'll move this comment to the other pull request.

**Comment by hkershaw-brown** [comment]
Can you put them on a list, since there might be a bunch. 
If they are minor edits (like removing a "new and only supports a subset of the models") can you put in a suggestion to get a second pair of eyes. 
We have to be a bit more careful that usual because we have readthedocs building off this branch.  So no late night /weekend pushes for those workaholics.

**Comment by timhoar** [comment]
Are we ready to update this branch and either merge it or kill it?

**Comment by nancycollins** [comment]
i was waiting to work on this until the docs, website and a formal release was announced.  if i'm supposed to be working on code again now, i'll finish this.  i have some of the suggested changes already implemented and a couple others to finish.  please advise.

File: assimilation_code/programs/obs_utils/obs_info.f90
Comment (nancycollins) [reply assimilation_code/programs/obs_utils/obs_info.f90:248]: agreed.  fixed in my repo and will be pushed soon.

File: assimilation_code/programs/obs_utils/obs_info.f90
Comment (nancycollins) [reply assimilation_code/programs/obs_utils/obs_info.f90:241]: this isn't the line that needs fixing (it's CSV format for ingest into matlab).  i have fixed the line at 244 that does need spaces in it.  (not committed yet but soon.)

File: assimilation_code/programs/obs_utils/obs_info.f90
Comment (nancycollins) [reply assimilation_code/programs/obs_utils/obs_info.f90:231]: i wanted to use identical formats for both regular obs and identity obs.

**Review by nancycollins — COMMENTED** [review]
i've found and compiled my version of this.  i will do some more testing and then push my changes to the public repo.

**Comment by nancycollins** [comment]
ok, the fixes are pushed and if approved i think they're ready to commit.  this fixes issue #186 (obs_info crashes with identity obs) and it lines up the output cleanly.

**Comment by timhoar** [comment]
I input a list of files and the run-time output was REALLY confusing:

```
--------------------------------------
 Starting ... at YYYY MM DD HH MM SS = 
                 2021  5 10 11 16 10
 Program obs_info
 --------------------------------------

  set_nml_output Echo NML values to log file only
 
 Assimilate_these_obs_types:
    RAW_STATE_VARIABLE
 Evaluate_these_obs_types:
    none
 Use the precomputed Prior Forward Operators for these obs types:
    none
 
  obs_info  Starting to process input sequence file:
  obs_info ...  obs_seq.out
 
 Totals for all obs types:
   Count:          600
.  First obs: 1601 Jan 01 00:00:00
.   Last obs: 1601 Feb 19 18:00:00
 ---------------------------------------------------------
 IDENTITY_OBSERVATIONS               600
.  First obs: 1601 Jan 01 00:00:00
.   Last obs: 1601 Feb 19 18:00:00
  obs_info  Starting to process input sequence file:
  obs_info ...  obs_seq.out.average
 
 Totals for all obs types:
   Count:          600
.  First obs: 1601 Jan 01 00:00:00
.   Last obs: 1601 Feb 19 18:00:00
 ---------------------------------------------------------
 RAW_STATE_VARIABLE                  600
.  First obs: 1601 Jan 01 00:00:00
.   Last obs: 1601 Feb 19 18:00:00
  obs_info  Starting to process input sequence file:
  obs_info ...  obs_seq.out.x
 
 Totals for all obs types:
   Count:          200
.  First obs: 1601 Jan 01 00:00:00
.   Last obs: 1601 Feb 19 18:00:00
```

I could not tell where the summary for one file ended and the next began ... and the delimeter line is (I think) in the **middle** of the output for that file.

**Comment by timhoar** [comment]
I changed this to a draft and am working on changing the output format. 
This might also be the time to create the matching   obs_info.rst   (which does not exist).  I am knocking out a version and will commit that as well. That will require modifying some of the navigational resources as well.

**Comment by nancycollins** [comment]
this was originally a one-off to output the obs info in CSV format for ingest into matlab so i could plot a year's worth of summary info about the observation seq files going into the reanalysis.  then i added formatting from another program that was a bit more human readable but i stopped there.  i think you should commit this pull request because it fixes the bug with identity obs, and then open another issue that asks this program to use print_obs_seq instead of trying to fix the formatting here as a one-off.  just my 2c.

**Comment by nancycollins** [comment]
exactly my point.  it should.

File: assimilation_code/programs/obs_utils/obs_info.nml
Comment (timhoar) [inline assimilation_code/programs/obs_utils/obs_info.nml:8]: These reflect the defaults in _obs_info.f90_

**Review by timhoar — APPROVED** [review]

**Comment by hkershaw-brown** [comment]
@timhoar are you working on updating the changelog.rst  and tag for this?

**Pull request merged**
